sdk/core/core-paging/review/core-paging.api.md

```diff
@@ -7,6 +7,12 @@
 // @public
 export function getPagedAsyncIterator<TElement, TPage = TElement[], TPageSettings = PageSettings, TLink = string>(pagedResult: PagedResult<TPage, TPageSettings, TLink>): PagedAsyncIterableIterator<TElement, TPage, TPageSettings>;
 
+// @public (undocumented)
+export type Paged<T> = {
+    value: T[];
+    nextLink?: string;
+};
```

Comment:
The return type of PagedResult.getPage is `{ page: TPage; nextPageLink?: TLink }` which is different from this Type, do you think we should do something to make them consistent ?

Reply:
We should either align or have good justification for why not. I would defer to doing what we're already doing in pagination libraries all else aside.

Reply:
Do you mean we should change the generated Paged<T> type so that it could be consistent with the return type of PagedResult.getPage ? But the Paged<T> comes from the cadl azure core, I wonder if this would cause some runtime error as this should be consistent with the return type of the service backend.

Reply:
The intent of my original comment is to encourage definition sharing, i.e. incorporating your type into the return type of `getPage` to eliminate duplication. However, I agree with [Brian's other comment](https://github.com/Azure/azure-sdk-for-js/pull/23906#discussion_r1061880072) that unless we have strong reasons to introduce those types in core, we shouldn't add them in the first place.

Reply:
The reason is kind of similar with LRO headers. If we don't add them into the core-paging library, we will end up with a lot of Models with similar structure. 
```
/** Paged collection of Project items */
export interface ProjectListOutput {
  /** The Project items on this page */
  value: Array<ProjectOutput>;
  /** The link to the next page of items */
  nextLink?: string;
}
/** Paged collection of Deployment items */
export interface DeploymentListOutput {
  /** The Deployment items on this page */
  value: Array<DeploymentOutput>;
  /** The link to the next page of items */
  nextLink?: string;
}
/** A collection of SupportedLanguage resources. */
export interface PagedSupportedLanguageOutput {
  /** The SupportedLanguage items on this page */
  value: Array<SupportedLanguageOutput>;
  /** The link to the next page of items */
  nextLink?: string;
}
/** A collection of TrainingConfigVersion resources. */
export interface PagedTrainingConfigVersionOutput {
  /** The TrainingConfigVersion items on this page */
  value: Array<TrainingConfigVersionOutput>;
  /** The link to the next page of items */
  nextLink?: string;
}
```
if we have the Paged<T> in the core-paging, we can just import it and the generated code would be 
```
/** Paged collection of Project items */
export type ProjectListOutput = Paged<ProjectOutput>;
/** Paged collection of Deployment items */
export type DeploymentListOutput = Paged<DeploymentOutput>;
/** A collection of SupportedLanguage resources. */
export type PagedSupportedLanguageOutput = Paged<SupportedLanguageOutput>;
/** A collection of TrainingConfigVersion resources. */
export type PagedTrainingConfigVersionOutput =
  Paged<TrainingConfigVersionOutput>;
```
This is the main reason I want to put it into the core-paging. If you both don't like the idea, maybe we could just add the ErrorModel into core-client and revert the change of core-paging and core-lro.

Reply:
I'm ok having these in core since presumably they will correspond 1:1 with the cadl primitives in Azure Core there, and so the days of proliferation of many different page models should be drawing to a close...

Reply:
Ok let's keep `Paged`.